signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> That definitely needs to happen somewhere, and right now it doesn’t. This is a bad fit, though, we should be verifying the user-specified digest even if no signatures exist or are required. So far I was thinking that the individual ImageSources which support digest references should do that in their GetManifests; it could probably be shared in copy.Image, OTOH it seems reasonable that this should be rejected for any use of that ImageSource.

I don't agree GetManifest(s) should do this check though. GetManifest just returns raw data, better to have callers verify digests, image sources are pretty low level, I wouldn't want to clutter them.